### ISSUE TITLE
brace-style stroustrup fails on switch statement

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#4004]
With the latest master `brace-style` with stroustrup option will fail on the following code:
```js
switch(foo) {
  case "bar": break;
}
```

error message returned by ESLint is:
```
1:1  error  Opening curly brace appears on the same line as controlling statement  brace-style
```

This is a result of #3904. @aparajita already provided a PR with a fix for this issue here: #4003 

P.S. I think error message is also incorrect and should point to the brace that is causing a problem

**Comment by nzakas** [comment]
It looks like #4003 deals with Allman style, not stroustrup as this issue reports. Are there two bugs?

**Comment by ilyavolodin** [comment]
#4003 fixes the same issue that #3904 fixed. The only difference is #3904 introduced a new issue reported here, and #4003 does not. It's basically all in this line of code: https://github.com/eslint/eslint/pull/4003/files#diff-224abf2dae2a4049db4b0ff3ae3e8d90R185

**Comment by nzakas** [comment]
So should this issue say "Allman" instead of "stroustrup"? And should #4003 be fixing this issue instead of #4002?

**Comment by lo1tuma** [comment]
We need a test for a switch statement with "stroustrup" style anyway, which isn’t added by #4003. So I think we should continue with this issue about "stroustrup".

**Comment by ilyavolodin** [comment]
Ehh.. It's a bit confusing, I know:-) Original issue was allman switch statements not being supported. #3904 fixed that, but broke stroustrup switch statements (`style === '1tbs'` vs `style!=='allman'`). #4003 should be renamed and commit message should be updated to point to this issue instead of #4002 . If @aparajita will be willing to do that, we can merge his commit, if not, we'll just need to resubmit with correct commit message.
Edit: @lo1tuma is right. So maybe we need a new PR anyways.

**Comment by aparajita** [comment]
It turns out there are more bugs in the allman switch testing. I suggest that I update my original PR with the new fixes and add the stroustrup tests to validate that the PR didn't break that.

**Comment by aparajita** [comment]
And at first look perhaps the other non-block statements like try/catch are also wrong with allman. Not to mention the positions reported for the errors are all wrong, they point to the keywords and not the offending braces.

**Comment by aparajita** [comment]
And the rule is using the deprecated context methods instead of SourceCode methods.

**Comment by aparajita** [comment]
I'm willing to fix all of this, but not if you want a separate issue and PR for every separate bug I found.

**Comment by nzakas** [comment]
We need an issue describing everything you're fixing. You can file one issue with multiple problems, just make sure you explain them clearly. Then you can do one PR to fix them all.

**Comment by aparajita** [comment]
Fine, as long as I only need one PR for all of the problems.

**Comment by nzakas** [comment]
Yup. Now what's the state of this issue? Was it fixed?

**Comment by aparajita** [comment]
The immediate problem was fixed by #4003.

**Comment by ilyavolodin** [comment]
This issue has been fixed. @aparajita Can you open a new issue describing all of the problems that you found with `brace-style` that way we can accept a single PR for all of them.
